sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -187,6 +191,8 @@ def _iter_next(self):
             self._message_iter = self._handler.receive_messages_iter()
         uamqp_message = next(self._message_iter)
         message = self._build_message(uamqp_message)
+        if self._auto_lock_renewer and not self._session:
+            self._auto_lock_renewer.register(self, message)
```

Comment:
I'm thinking about `ReceiveAndDelete` mode in normal receiver.

should we make this a no-op in that case or raise Error saying that "AutoLockRenewer" cannot be applied to ReceivedAndDelete mode in a non-sessionful case?

Reply:
It's actually quite an interesting question.  Throwing a Value/TypeError would give much more immediate feedback, since the renewer isn't going to be doing anything useful, but it also means a user has to twiddle more params to swap between the two when "technically" passing the renewer will "fail cleanly" (e.g. it'll try to renew, fail to, put an error in the callback handler, then go on happily.)  I don't have a strong opinion over the cuff but will try to form one as I heat up my lunch.  Did you have a preference?

Reply:
from OOB: other languages have Noop since it's opt-out, but given our decision to opt-in due to python idioms and our non-processor call patterns, we're having it fail fast (as general "good practice") to try and be principled and give good user feedback.